File: sort/c/src/sort/sort.c
// sort/c/src/sort/sort.c

#include "sort.h"

/*
------------------------------------------------------------------------
    sort  -  Common functions for sorting
------------------------------------------------------------------------
*/

// swap the values in the given array and indeces i and j.
void swap(array *arptr, size_t i, size_t j) {
    float tmp = arptr->ar[i];
    arptr->ar[i] = arptr->ar[j];
    arptr->ar[j] = tmp;
}

File: sort/go/src/sort/sort.go
// sort/go/src/sort/sort.go

package sort

// ------------------------------------------------------------------------
//     sort  -  Common functions for sorting
// ------------------------------------------------------------------------

// swap the values in the given array and indeces i and j.
func swap(ar *[]float32, i int, j int) {
	(*ar)[i], (*ar)[j] = (*ar)[j], (*ar)[i]
}
